### DISCUSSION TITLE
Add option for Ctrl + Drag Mouse Wheel Up / Down to change 2D zoom level

### DISCUSSION THREAD

**Discussion by JJPSeno** [godotengine/godot-proposals#11446] [2D]
There are quite a few ways to change the zoom level in the 2D Inspector that I am aware of:
The zoom-in and zoom-out buttons on the scene preview - 
![Button Zooming](https://github.com/user-attachments/assets/867b6176-4161-4115-b774-bb7dff26f47a)
The mouse wheel -
![Mouse Wheel Zooming](https://github.com/user-attachments/assets/6efa2fe5-57a5-459c-858b-d8f0af4c4d9d)
Keyboard shortcuts -
![Shortcuts](https://github.com/user-attachments/assets/dc73d219-9a64-4070-a566-564f05af38ea)

I'd like to propose that we have another option as well, which has been implemented in something like GIMP:
![Gimp Zooming](https://github.com/user-attachments/assets/05006822-46b8-49ae-ac2a-5a4536ce0d17)
Which is to allow Ctrl + Clicking and Dragging the Mouse Wheel Up / Down (not scrolling) to change 2D zoom level

And I believe Godot also already has this feature, but only for the 3D Inspector and I suppose the bottom line is I'd also love for this feature to be ported to the 2D Inspector:
![Godot 3D Zooming](https://github.com/user-attachments/assets/44dadcbc-0f00-4a24-8db4-91bca56394b6)

**Comment by Calinou** [comment]
<kbd>Ctrl + Mouse wheel</kbd> scrolls vertically in the 2D editor, while using the mouse wheel alone zooms. Such an editor setting would probably swap the two behaviors together to match GIMP and web browsers.

PS: I noticed <kbd>Shift + Mouse wheel</kbd> does nothing. We should probably make it scroll horizontally regardless of this setting to match web browsers.

**Reply by JJPSeno** [reply]
For `ctrl + mouse wheel` do you mean scrolling the mouse wheel up and down? I see the behavior you described by scrolling the mouse wheel, but I actually meant clicking and holding the mouse wheel down and dragging it (which pans the camera). Is there still a conflict with the setting in this case?

**Reply by SethFalco** [reply]
@Calinou 

> PS: I noticed <kbd>Shift + Mouse wheel</kbd> does nothing. We should probably make it scroll horizontally regardless of this setting to match web browsers.

I was actually just working on a proposal to request this for the 2D editor! I'll refrain for now you've mentioned it here, but let me know if I should post it anyway. I use that to scroll horizontally across many applications, so it would be wonderful to have in Godot by default too.

<details>
<summary>WIP proposal for <kbd>SHIFT</kbd> + <kbd>Scroll Wheel</kbd> to horizontally scroll 2D editor before seeing this comment.</summary>

## Shift + Scroll Wheel to horizontally scroll canvas in 2D view

Conventionally, many applications (namely text editors and image manipulation software) allow the following controls:

* <kbd>Scroll ­Wheel</kbd> — Vertical scrolling
* <kbd>CTRL</kbd> + <kbd>Scroll Wheel</kbd> — Zoom
* **<kbd>SHIFT</kbd> + <kbd>Scroll Wheel</kbd> — Horizontal scrolling** ← I'd like to focus on this one.

I'm aware that the 2D view in Godot does zoom and vertical scrolling differently, but that's fine as this feature request focuses on horizontal scrolling specifically.

Would it be fine to add horizontal scrolling to the 2D view, using <kbd>SHIFT</kbd> + <kbd>Scroll Wheel</kbd>? 🤔 

</details>

**Reply by Calinou** [reply]
As of 4.4.1, this appears to already work in both the 2D editor and the visual shader editor, but you need to hold <kbd>Ctrl</kbd> *and* <kbd>Shift</kbd> for it to work since using the mouse wheel alone will zoom by default.

You can change whether the mouse wheel alone zooms or pans in the Editor Settings (`editors/panning/2d_editor_panning_scheme`, `editors/panning/animation_editors_panning_scheme`, `editors/panning/sub_editors_panning_scheme`).

**Reply by SethFalco** [reply]
Ahh I see, I never tried <kbd>Ctrl</kbd> + <kbd>Shift</kbd>. Thanks for clarifying!
